### PR TITLE
Feature/pdct 1299 encode a configuration token for custom apps

### PR DESCRIPTION
**Pull request by odrakes-cpr**
# Description

Create function to configure and encode a list of corpora ids into a jwt token. As part of our work creating custom apps we want to ensure that the appropriate data flows through to the correct corpus, this is one of the first steps in setting up that process.

Please include:

- a summary of the changes
- links to any related issue/ticket
- any additional relevant motivation and context
- details of any dependency updates that are required for this change

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Refactor legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

**Review by diversemix — COMMENTED** [review]
Probably worth a simple unit-test and a README.md ? ... but looking good :+1:

**Comment by diversemix** [comment]
I tested with 
```
python -c "from app.core.custom_app import create_configuration_token ; print(create_configuration_token('a,b,c'))"
```

got the following 
```
Creating custom app configuration token that expires on Wed 26 July 2034 at 13:56:58:814926 for the following corpora: ['a', 'b', 'c']
eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhbGxvd2VkX2NvcnBvcmFfaWRzIjpbImEiLCJiIiwiYyJdLCJleHAiOjIwMzc1MzUwMTh9.i1RAjvHwoe-t7vmsGbcIXJGKUnZI0FlswXocdiHxpD8
```

Which then decoded to :
```
{
  "allowed_corpora_ids": [
    "a",
    "b",
    "c"
  ],
  "exp": 2037535018
}
```

File: app/core/custom_app.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    corpora_ids = allowed_corpora.split(",")
+    corpora_ids.sort()
+
+    msg = "Creating custom app configuration token that expires on "
+    msg += f"{expire.strftime('%a %d %B %Y at %H:%M:%S:%f')} "
+    msg += f"for the following corpora: {corpora_ids}"
+    print(msg)
+
+    to_encode = {"allowed_corpora_ids": corpora_ids, "exp": expire}
+    return jwt.encode(to_encode, SECRET_KEY, algorithm=ALGORITHM)
```
[the remark this answers is not in the document]
Comment (diversemix) [reply app/core/custom_app.py:58]: Would also be good to include the "issuer" , "Subject" and "audience" - see https://en.wikipedia.org/wiki/JSON_Web_Token#Standard_fields

[the remark this answers is not in the document]
Comment (katybaulch) [reply app/core/custom_app.py:63]: Have added `iat`, "issuer" , "Subject" and "audience" to come in the next PR when we integrate this into an endpoint to generate the token from (as we'll change from taking a string input to JSON -- this seems like a nice logical place to split the PRs to keep them small)

**Comment by katybaulch** [comment]
> Probably worth a simple unit-test and a README.md ? ... but looking good 👍

Have added tests - Osneil has the readme in his branch :)

**Review by Odrakes1992 — APPROVED** [review]
V nice tests

**Review by diversemix — APPROVED** [review]
Nice foundation :+1:

**Pull request merged**